lib/utils/urls.js

```diff
@@ -15,18 +15,24 @@ export function addPort(uri, location) {
 }
 
 /**
- * add protocol to uri if it needs it
- * @param {string} uri
- * @param {object} [location] to stub for testing
- * @returns {string}
- */
-export function addProtocol(uri, location) {
-  location = location || /* istanbul ignore next: can't stub window.location */ window.location;
-  const hasProtocol = uri.indexOf(location.protocol) === 0;
+* Check url string for protocol and if it doesn't have one, add the protocol of the current url
+* @param {string} url
+* @param {object} location - only passed in testing
+* @return {string}
+*/
+export function addProtocol(url, location) {
+  const pattern = /^((http|https):\/\/)/;
```

Comment:
This can be set as a global at the top of the file so we don't need to create this property every time the `addProtocol` function is called